zerver/tests/test_send_email.py

```diff
@@ -70,39 +74,185 @@ def test_limited_to_length(self) -> None:
         # Test the new connection case
         with mock.patch.object(EmailBackend, "open", return_value=True):
             backend = initialize_connection(None)
             self.assertTrue(isinstance(backend, EmailBackend))
 
-        backend = mock.MagicMock(spec=SMTPBackend)
-        backend.connection = mock.MagicMock(spec=SMTP)
-
-        self.assertTrue(isinstance(backend, SMTPBackend))
-
-        # Test the old connection case when it is still open
-        backend.open.return_value = False
-        backend.connection.noop.return_value = [250]
-        initialize_connection(backend)
-        self.assertEqual(backend.open.call_count, 1)
-        self.assertEqual(backend.connection.noop.call_count, 1)
-
-        # Test the old connection case when it was closed by the server
-        backend.connection.noop.return_value = [404]
-        backend.close.return_value = False
-        initialize_connection(backend)
-        # 2 more calls to open, 1 more call to noop and 1 call to close
-        self.assertEqual(backend.open.call_count, 3)
-        self.assertEqual(backend.connection.noop.call_count, 2)
-        self.assertEqual(backend.close.call_count, 1)
-
-        # Test backoff procedure
-        backend.open.side_effect = OSError
-        with self.assertRaises(OSError):
+        backend = PersistentSMTPEmailBackend()
+        backend.open()
+        self.assertEqual(backend.opened_at, timezone_now())
+
+        with (
+            mock.patch.object(backend, "_open", wraps=backend._open) as _open_call,
+            mock.patch.object(backend, "close", wraps=backend.close) as close_call,
+        ):
+            backend.connection = mock.MagicMock(spec=SMTP)
+            backend.connection.noop.return_value = [250]
+            initialize_connection(backend)
+
+            self.assertEqual(_open_call.call_count, 1)
+            self.assertEqual(backend.connection.noop.call_count, 1)
+            close_call.reset_mock()
+            _open_call.reset_mock()
+
+            # Test the old connection case when it was closed by the server
+            backend.connection.noop.return_value = [404]
+            initialize_connection(backend)
+
+            # 2 calls to open and 1 call to close
+            self.assertEqual(_open_call.call_count, 2)
+            self.assertEqual(close_call.call_count, 1)
+            close_call.reset_mock()
+            _open_call.reset_mock()
+
+            # Test backoff procedure
+            _open_call.side_effect = OSError
+            with self.assertRaises(OSError):
+                initialize_connection(backend)
+            # 3 calls to open as we try 3 times before giving up
+            self.assertEqual(_open_call.call_count, 3)
+
+    @time_machine.travel(timezone_now(), tick=False)
+    @override_settings(EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=0)
+    def test_max_connection_lifetime_zero(self) -> None:
+        backend = PersistentSMTPEmailBackend()
+        backend.open()
```

Comment:
We should do this step inside of where we set up our mocks.  We can thus test how things work from an initial state -- which should be 0 noops, 0 closes, and 1 open.